ArmoniK.Extension.CSharp.DllCommon/SdkTaskHandler.cs

```diff
@@ -0,0 +1,240 @@
+  ///   The data required to compute the task. The key is the name defined by the client, the value is the raw data.
+  /// </summary>
+  public IReadOnlyDictionary<string, byte[]> Inputs { get; }
+
+  /// <summary>
+  ///   Result blob ids by name defined by the client.
+  /// </summary>
+  public IReadOnlyDictionary<string, string> Outputs { get; }
+
+  /// <summary>
+  ///   The configuration parameters for the interaction with ArmoniK.
+  /// </summary>
+  public Configuration Configuration
+    => taskHandler_.Configuration;
+
+  /// <summary>
+  ///   Decode a dependency from its raw data
+  /// </summary>
+  /// <param name="name">The input name defined by the client</param>
+  /// <returns>The decoded string</returns>
+  public string GetStringDependency(string name)
+    => Encoding.UTF8.GetString(Inputs[name]);
+
+  /// <summary>
+  ///   Decode a dependency from its raw data
+  /// </summary>
+  /// <param name="name">The input name defined by the client</param>
+  /// <returns>The decoded integer</returns>
+  public int GetIntDependency(string name)
+    => BitConverter.ToInt32(Inputs[name]);
+
+  /// <summary>
+  ///   Decode a dependency from its raw data
+  /// </summary>
+  /// <param name="name">The input name defined by the client</param>
+  /// <returns>The decoded double</returns>
+  public double GetDoubleDependency(string name)
+    => BitConverter.ToDouble(Inputs[name]);
+
+  /// <summary>This method allows to create subtasks.</summary>
+  /// <param name="tasks">Lists the tasks to submit</param>
+  /// <param name="taskOptions">The task options. If no value is provided, will use the default session options</param>
+  /// <param name="cancellationToken">
+  ///   Token used to cancel the execution of the method.
+  ///   If null, the cancellation token of the task handler is used
+  /// </param>
+  /// <returns></returns>
+  public async Task<CreateTaskReply> CreateTasksAsync(IEnumerable<TaskRequest> tasks,
+                                                      TaskOptions?             taskOptions       = null,
+                                                      CancellationToken?       cancellationToken = null)
+    => await taskHandler_.CreateTasksAsync(tasks,
+                                           taskOptions,
+                                           cancellationToken)
+                         .ConfigureAwait(false);
+
+  /// <summary>
+  ///   NOT IMPLEMENTED
+  ///   This method is used to retrieve data available system-wide.
+  /// </summary>
+  /// <param name="key">The data key identifier</param>
+  /// <param name="cancellationToken">
+  ///   Token used to cancel the execution of the method.
+  ///   If null, the cancellation token of the task handler is used
+  /// </param>
+  /// <returns></returns>
+  public async Task<byte[]> RequestResource(string             key,
+                                            CancellationToken? cancellationToken = null)
+    => await RequestResource(key,
```

Comment:
```suggestion
    => await taskHandler_.RequestResource(key,
```

otherwise isn't it calling yourself ?